### PR TITLE
Add Optuna and Improve Doc-strings for Auto3Dseg

### PR DESCRIPTION
**Pull request by mingxin-zheng**
### Description

Improve docstring and Optuna Integration(#5136 )

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [x] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/auto3dseg/hpo_gen.py [hunk 1]
```diff
@@ -227,3 +229,164 @@ def run_algo(self, obj_filename: str, output_folder: str = ".", template_path=No
...
+    def set_score(self, acc):
+        self.acc = acc
+
+    def set_trial(self, trial):
+        self.trial = trial
+
+    def print_bundle_algo_instruction(self):
+        pass
+
+    def __call__(self, trial, obj_filename: str, output_folder: str = ".", template_path=None):
```
Comment (Nic-Ma) [inline monai/apps/auto3dseg/hpo_gen.py:321]: Could you please help add the doc-string here?

Thanks.

File: monai/apps/auto3dseg/hpo_gen.py [hunk 2]
```diff
@@ -227,3 +229,164 @@ def run_algo(self, obj_filename: str, output_folder: str = ".", template_path=No
...
+            warn("Optuna is not detected. The code will continue to run without Optuna.")
+            return {}
+
+    def set_score(self, acc):
+        self.acc = acc
+
+    def set_trial(self, trial):
+        self.trial = trial
+
+    def print_bundle_algo_instruction(self):
```
[the remark this answers is not in the document]
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/hpo_gen.py]: I forgot to delete this. It tries to overload the function with the same name from NNIGen

File: monai/apps/auto3dseg/hpo_gen.py (hunk 1 above)
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/hpo_gen.py:320]: Added comments. Thanks

**Comment by Nic-Ma** [comment]
Hi @wyli ,

I think we also need to add the optional dependency to this file:
https://github.com/Project-MONAI/MONAI/blob/dev/requirements-dev.txt
But the contribution guide doesn't include it, can @mingxin-zheng also update the contribution guide doc to include it?

Thanks.

**Review by Nic-Ma — APPROVED** [review]
Looks good to me except for the:
https://github.com/Project-MONAI/MONAI/blob/dev/requirements-dev.txt

Thanks.

**Comment by mingxin-zheng** [comment]
> Looks good to me except for the: https://github.com/Project-MONAI/MONAI/blob/dev/requirements-dev.txt
> 
> Thanks.

Thanks @Nic-Ma . I have updated the template package and the requirement-dev as well.

**Pull request merged**